src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java

```diff
@@ -47,17 +47,32 @@
   static final String RULE_MESSAGE = "Always use HTL style of comments";
 
   private static final Pattern CONDITIONAL_COMMENT_PATTERN = Pattern
-      .compile("<!--\\[if.*!\\[endif\\]-->");
+      .compile("<!--\\[.*]-->");
+
+  private static final Pattern ESI_COMMENT_PATTERN = Pattern
+      .compile("<!--esi.*-->");
 
   @Override
   public void comment(CommentNode node) {
     String code = node.getCode();
-    if (!Syntax.isSightlyComment(code) && !isConditionalComment(code)) {
+    if (!Syntax.isSightlyComment(code) && !isException(code)) {
       createViolation(node.getStartLinePosition(), RULE_MESSAGE);
     }
   }
 
+  private boolean isException(String code) {
+    return isConditionalComment(code) || isESIComment(code);
```

Comment:
I would rather create static list of patterns 'exceptions' 

`EXCEPTIONS = Lists.newArrayList(CONDITIONAL_COMMENT_PATTERN, ESI_COMMENT_PATTERN)`

the use:

```
exceptions.stream()
.map(pattern -> pattern.matcher(code))
.anyMatch(Matcher::matches);
```

It would be easier to add more regex patterns.

Reply:
Done, looks nice.